Foundation/Process.swift

```diff
@@ -241,7 +241,11 @@ open class Process: NSObject {
         if let env = environment {
             let nenv = env.count
             envp = UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>.allocate(capacity: 1 + nenv)
-            envp.initialize(from: env.map { strdup("\($0)=\($1)") })
+            let evars = env.map { (arg) -> UnsafeMutablePointer<Int8>? in
+                let (key, value) = arg
+                return UnsafeMutablePointer(strdup("\(key)=\(value)"))
+            }
+            envp.initialize(from: evars, count: 1 + nenv)
```

Comment:
hmm there is a size mismatch, nenv is the count of elements in env, so the initialization will be reading past the mapped value.

The expectation here is the envp has a "null terminator"